cpp/ql/lib/semmle/code/cpp/dataflow/internal/TaintTrackingUtil.qll

```diff
@@ -166,6 +166,11 @@ private predicate noFlowFromChildExpr(Expr e) {
   e instanceof FieldAccess
 }
 
+/**
+ * Holds if there is a data flow from `exprIn` to `exprOut` in the code.
+ * The predicate also handles taint flows, where the data flow is from a tainted source to a sink.
+ * It checks if there is a taint flow from `exprIn` to `exprOut` by considering the same cases as above.
+ */
```

Comment:
This explanation is a bit mechanical.  What I want to know is where `exprToExprStep` fits in to the data flow construction.